File: Longest Palindrome in a String[15-June-2023]/Solution.cpp
//  >> C++ CODE <<

string longestPalin(const string& S) {
        int st = 0;
        int end = 0;
        int i = 0;
        int n = S.length();
        
        while (i < n) {
            int j = i - 1;
            int k = i + 1;
            
            while (j >= 0 && k < n && S[j] == S[k]) {
                k++;
                j--;
            }
            
            if (k - j + 1 > end - st + 1) {
                end = k;
                st = j;
            }
            
            j = i;
            k = i + 1;
            
            while (j >= 0 && k < n && S[j] == S[k]) {
                k++;
                j--;
            }
            
            if (k - j + 1 > end - st + 1) {
                end = k;
                st = j;
            }
            
            i++;
        }
        
        return S.substr(st + 1, end - st - 1);
}

File: Longest Palindrome in a String[15-June-2023]/Solution.py
#  >> PYTHON CODE <<

def longestPalin(self, S):
        st=0
        end=0
        i=0
        n=len(S)
        while(i<n):
            j=i-1
            k=i+1
            
            while(j>=0 and k<n and S[j]==S[k]):
                k+=1
                j-=1
            
            if(k-j+1>end-st+1):
                end=k
                st=j
            
            j=i
            k=i+1
            
            while(j>=0 and k<n and S[j]==S[k]):
                k+=1
                j-=1
            
            if(k-j+1>end-st+1):
                end=k
                st=j
            
            i+=1
            
        
        return S[st+1:end]